### PR TITLE
refractoring neutronics (again)

### PR DESCRIPTION
**Pull request by shimwell**
## Proposed changes

This is a refactoring of the neutronics which tidies things up and. Features added include:

- ability to speed up graveyard_creation by specifying the ```graveyard_size```
- ability to speed up mesh creation and extra control f the location of the regular mesh using ```mesh_2d_corners``` and ```mesh_3d_corners```
- movement of export_h5m to Shape and Reactor classes
- ability to skip production of h5m files when simulating
- ability to use preexisting stp files instead of making them
- changed graveyard to a lower case (compatible with shift)
- reduced the complexity of methods by separating them and moving some methods to utils functions
- added support for sector models
- fixed a bug in the submersion reactor where the inner blanket overlapped the divertor

TODO
add tests

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [x] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
@Shimwell should we solve #752 in this PR ?

**Comment by shimwell** [comment]
> @Shimwell should we solve #752 in this PR ?

Yep, I shall add the extra tests soon and we shall see how much time it ends up taking. Then we can move it to wherever you like

**Comment by shimwell** [comment]
Not sure why the circle ci test for tests/test_parametric_neutronics is failing. It works locally

**Comment by RemDelaporteMathurin** [comment]
> Not sure why the circle ci test for tests/test_parametric_neutronics is failing. It works locally

are you using the exact same environment ?

**Comment by shimwell** [comment]
Looks like the CI (circle ci at least) is passing the tests. I shall add a few extra tests to sort out that drop in code coverage.

**Comment by shimwell** [comment]
Code coverage appears to be missing a few lines that are covered by the tests

**Comment by shimwell** [comment]
I think it might be time to merge this if there are no objections

**Comment by shimwell** [comment]
the tests cover-cov appears to be missing are covered by the 
    def test_find_materials_in_h5_file(self):
    def test_make_watertight_cmd_with_example_dagmc_file(self):
in the test_neutronics_utils.py file

**Comment by RemDelaporteMathurin** [comment]
> the tests cover-cov appears to be missing are covered by the
> def test_find_materials_in_h5_file(self):
> def test_make_watertight_cmd_with_example_dagmc_file(self):
> in the test_neutronics_utils

How is that possible ? Seems weird

**Comment by shimwell** [comment]
> > the tests cover-cov appears to be missing are covered by the
> > def test_find_materials_in_h5_file(self):
> > def test_make_watertight_cmd_with_example_dagmc_file(self):
> > in the test_neutronics_utils
> 
> How is that possible ? Seems weird

Ah I might have found the problem. I thik it is missing a --cov append in the circle yml

**Comment by shimwell** [comment]
code coverage is +0.04% :tada:

That patch is still low but perhaps that is acceptable?

File: tests/test_neutronics_utils.py [hunk 1]
```diff
@@ -103,3 +125,259 @@ def incorrect_viewplane():
...
+        source.space = openmc.stats.Point((1, 2, 3))
+        source.energy = openmc.stats.Discrete([14e6], [1])
+        source.angle = openmc.stats.Isotropic()
+
+        source_file = paramak.neutronics_utils.create_inital_particles(
+            source=source,
+            number_of_source_particles=10
+        )
+
+        assert "initial_source.h5" == source_file
```
Comment (shimwell) [inline tests/test_neutronics_utils.py]:
```suggestion
        assert source_file == "initial_source.h5"
```

**Review by billingsley-john — APPROVED** [review]
All looks good!

**Pull request merged**